api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2294,6 +2297,182 @@ public void copyObject(String bucketName, String objectName, String destBucketNa
+                             List<CopySource> listSourceObjects)
+          throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+          InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+          InternalException, InvalidArgumentException {
+
+    composeObject(destinationBucketName, destintionObjectName, null, null, listSourceObjects);
+
+  }
+
+
+  /**
+   * ComposeObject - It creates an object using server-side copying of
+   * existing objects. It takes a list of source objects and concatenates
+   * them into a new object using only server-side copying operations
+   *
+   * </p>
+   * <b>Example:</b><br>
+   *
+   * <pre>
+   * {@code minioClient.composeObject(String destinationBucketName, String destintionObjectName,
+   *       ServiceConfigurationError sse, Map<String,String> userMetaData, List<CopySource> listSourceObjects;}
+   * </pre>
+   *
+   * @param destinationBucketName
+   *          Destination Bucket to be created upon compose.
+   * @param destintionObjectName
+   *          Destination Object to be created upon compose.
+   * @param sse
+   *          Server Side Encryption.
+   * @param userMetaData
+   *          User Meta data.
+   * @param listSourceObjects
+   *          List of Source Objects used to compose Object
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   */
+
+
+  public void  composeObject(String destinationBucketName, String destintionObjectName, ServerSideEncryption sse,
+                             Map<String,String> userMetaData, List<CopySource> listSourceObjects)
+          throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+          InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+          InternalException, InvalidArgumentException {
+
+    if (listSourceObjects.size() < 1 || listSourceObjects.size() > MAX_MULTIPART_COUNT ) {
+      throw new InvalidArgumentException("There must be as least one and up to 10000 source objects.");
```

Comment:
you could use `MAX_MULTIPART_COUNT` than `10000`

Reply:
done.